### PR TITLE
initial commit of Parse-DbaCsvMetadata

### PR DESCRIPTION
**Pull request by barticsoka**
This was previously called  Get-CsvMetadata

Started synopsis, (still not complete...)
advanced command BEGIN, PROCESS, END added
pscustomobject is now in use
added bits and bobs... little tidy up

Verbose output sort of completed.
Few debug checkpoints available.
Includes some level of error handling.
E.g:
image

returns PS object with vital metadata which then can be parsed.
E.g.:
image

### CODE REVIEW

**Comment by dansqldba** [comment]
You're trying to add a file to functions and the root of the module. Remove the file for your cmdlet from the root of the module then commit to the fr3ky/dbatools/development branch again, which will update this PR.

File: functions/Parse-DbaCsvMetadata.ps1 [hunk 1]
```diff
@@ -0,0 +1,360 @@
+﻿function Get-UserPrompt {
+    param (
```
Comment (ctrlbold) [inline functions/Parse-DbaCsvMetadata.ps1]: Please move this function to your BEGIN block. All  files should start with your initial function name and be self contained (except for shared functions); please take a look at other commands such as [Copy-SqlLogin.ps1](https://github.com/sqlcollaborative/dbatools/blob/master/functions/Copy-SqlLogin.ps1) as an example. Placing functions in the BEGIN block makes them private to your script.

File: functions/Parse-DbaCsvMetadata.ps1 [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    .PARAMETER SpeedTest
+
+    If set, it will read the entire file and outputs read / time statistics.
+
+
+    .NOTES 
+
+    Original Author: You (@YourTwitter, Yourblog.net)
+
+
```
Comment (ctrlbold) [inline functions/Parse-DbaCsvMetadata.ps1]: This is the right stuff :) But the paste somehow added way more line breaks than necessary to this whole section, starting from SYNOPSIS. Please condense.

File: functions/Parse-DbaCsvMetadata.ps1 [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+                #  [long]      64-bit signed integer
+                #  [bool]      Boolean True/False value
+                #  [decimal]   A 128-bit decimal value
+                #  [single]    Single-precision 32-bit floating point number
+                #  [double]    Double-precision 64-bit floating point number
+                #  [DateTime]  Date and Time
+                #  [string]    Fixed-length string of Unicode characters
+                
+                $hash = [pscustomobject]@{
+                    NumberedHeader = $HeaderNumber
```
Comment (ctrlbold) [inline functions/Parse-DbaCsvMetadata.ps1]: Yeahhh :D

File: functions/Parse-DbaCsvMetadata.ps1 [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+        if ($abort)
+        {
+            Write-Verbose "[csv file] No CSV file selected, aborted."
+            return
+        }
+        return $MetaDataCollection
+    }
+}
+
+# This is a test output
```
Comment (ctrlbold) [inline functions/Parse-DbaCsvMetadata.ps1]: Please move or remove. Comments should be either in the top block or in the comments of the PR.

**Comment by ctrlbold** [comment]
Hey @barticsoka,
Are there any plans to make the required changes within the next month or two? If not, I'll close this PR and you can resubmit.

**Comment by ctrlbold** [comment]
Till next time!

**Pull request closed without merging**